Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift

```diff
@@ -222,8 +222,13 @@ extension ExperienceStateMachine {
             switch self {
             case .continuation(let action):
                 try machine.transition(action)
-            case let .presentContainer(experience, stepIndex, package):
-                executePresentContainer(machine: machine, experience: experience, stepIndex: stepIndex, package: package)
+            case let .presentContainer(experience, stepIndex, package, actions):
+                machine.actionRegistry.enqueue(actionModels: actions) {
```

Comment:
this is where the pre-step actions execute before the container presentation on line 227, which happens after action completion.